packages/ember-data/lib/system/model/model.js

```diff
@@ -34,6 +34,29 @@ function extractPivotName(name) {
   );
 }
 
+// Like Ember.merge, but instead returns a list of keys
+// for values that fail a strict equality check
+// instead of the original object.
+function mergeAndReturnChangedKeys(original, newInfo) {
+  var ret = [];
```

Comment:
Tiny nit-pick: can you name this something meaningful (to make it clearer when I end up stepping the code while in *brain-fried* mode)?  Something like `changedProperties` would make this much clearer...